soco/discovery.py

```diff
@@ -123,18 +158,19 @@ def discover(timeout=1, include_invisible=False, interface_addr=None):
         # X-RINCON-HOUSEHOLD: Sonos_7O********************R7eU
 
         if response:
-            data, addr = _sock.recvfrom(1024)
-            _LOG.debug('Received discovery response from %s: "%s"', addr, data)
-            if b"Sonos" not in data:
-                continue
-
-            # Now we have an IP, we can build a SoCo instance and query that
-            # player for the topology to find the other players. It is much
-            # more efficient to rely upon the Zone Player's ability to find
-            # the others, than to wait for query responses from them
-            # ourselves.
-            zone = config.SOCO_CLASS(addr[0])
-            if include_invisible:
-                return zone.all_zones
-            else:
-                return zone.visible_zones
+            for _sock in response:
+                data, addr = _sock.recvfrom(1024)
+                _LOG.debug(
+                    'Received discovery response from %s: "%s"', addr, data
+                )
+                if b"Sonos" in data:
+                    # Now we have an IP, we can build a SoCo instance and query
+                    # that player for the topology to find the other players.
+                    # It is much more efficient to rely upon the Zone
+                    # Player's ability to find the others, than to wait for
+                    # query responses from them ourselves.
+                    zone = config.SOCO_CLASS(addr[0])
+                    if include_invisible:
+                        return zone.all_zones
```

Comment:
I think that's the point - get all the systems and then prompt the user for which they want to use (which is what the Kodi addon does).  Otherwise where is the control for your "Preferred System".  I understand it may not be the way things work at the moment - but I see no need not to code for this sort of thing now.